assignments/prototypes.js

```diff
@@ -102,9 +135,80 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
 
-  // Stretch task: 
-  // * Create Villian and Hero constructor functions that inherit from the Humanoid constructor function.  
-  // * Give the Hero and Villians different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villian and one a hero and fight it out with methods!
\ No newline at end of file
+
+// Stretch task:
+// * Create Villian and Hero constructor functions that inherit from the Humanoid constructor function.
+// * Give the Hero and Villians different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
+// * Create two new objects, one a villian and one a hero and fight it out with methods!
+
+// Villian
+function Villian(greatGrandchildAttributes) {
+    Humanoid.call(this, greatGrandchildAttributes);
+}
+
+Villian.prototype = Object.create(Humanoid.prototype);
+
+Villian.prototype.maulsAttack = function () {
+    aragorn.hp -= 5;
+    if (witchKing.hp <= 0) {
+        return aragorn.destroy();
+    }
+    return `Remove 5 HP from hero. The life of ${aragorn.name} is ${aragorn.hp}`;
+};
```

Comment:
It's be better to use `this` and pass in the attacked as an object in order to make this reusable:
```javascript
Villian.prototype.maulsAttack = function (attacked) {
    attacked.hp -= 5;
    if (this.hp <= 0) {
        return attacked.destroy();
    }
    return `Remove 5 HP from hero. The life of ${attacked.name} is ${attacked.hp}`;
};
```

Reply:
The same applies to the `Hero`'s class attack method.

Reply:
I abstracted to use attacked but was still unable to use `this`. It breaks the if statement.